plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadata.java

```diff
@@ -0,0 +1,613 @@
+import io.trino.spi.connector.MaterializedViewFreshness;
+import io.trino.spi.connector.MaterializedViewNotFoundException;
+import io.trino.spi.connector.SchemaTableName;
+import io.trino.spi.connector.TableNotFoundException;
+import io.trino.spi.security.TrinoPrincipal;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeId;
+import io.trino.spi.type.TypeManager;
+import io.trino.spi.type.TypeOperators;
+import io.trino.spi.type.TypeSignature;
+import io.trino.testing.TestingConnectorSession;
+import io.trino.tests.BogusType;
+import org.apache.iceberg.AppendFiles;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DeleteFiles;
+import org.apache.iceberg.ExpireSnapshots;
+import org.apache.iceberg.HistoryEntry;
+import org.apache.iceberg.ManageSnapshots;
+import org.apache.iceberg.ManifestFile;
+import org.apache.iceberg.OverwriteFiles;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.ReplacePartitions;
+import org.apache.iceberg.ReplaceSortOrder;
+import org.apache.iceberg.RewriteFiles;
+import org.apache.iceberg.RewriteManifests;
+import org.apache.iceberg.Rollback;
+import org.apache.iceberg.RowDelta;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableScan;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.UpdateLocation;
+import org.apache.iceberg.UpdatePartitionSpec;
+import org.apache.iceberg.UpdateProperties;
+import org.apache.iceberg.UpdateSchema;
+import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.LocationProvider;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static io.trino.plugin.iceberg.TrinoHiveCatalog.DEPENDS_ON_TABLES;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.testcontainers.shaded.com.google.common.base.Preconditions.checkArgument;
+
+public class TestIcebergMetadata
+{
+    private static final CatalogSchemaTableName TESTING_MATERIALIZED_VIEW_NAME =
+            new CatalogSchemaTableName("test_catalog_name", "test_schema_name", "test_table_name");
+
+    private IcebergMetadata testingIcebergMetadata;
+    private TestingTrinoCatalog testingTrinoCatalog;
+
+    @BeforeClass
+    public void setup()
+    {
+        testingTrinoCatalog = new TestingTrinoCatalog();
+        testingIcebergMetadata = new IcebergMetadata(new TestingTypeManager(), jsonCodec(CommitTaskData.class), testingTrinoCatalog);
+    }
+
+    @AfterMethod
```

Comment:
CI says

```
2021-09-01T06:02:41.6255415Z FATAL: io.trino.testng.services.ReportMultiThreadedBeforeOrAfterMethod: Failed to process [TestClass name=class io.trino.plugin.iceberg.TestIcebergMetadata]: 
2021-09-01T06:02:41.6258743Z java.lang.RuntimeException: Test class io.trino.plugin.iceberg.TestIcebergMetadata should be annotated as @Test(singleThreaded=true), if it contains mutable state as indicated by public void io.trino.plugin.iceberg.TestIcebergMetadata.resetTestDependencies()
2021-09-01T06:02:41.6263476Z 	at io.trino.testng.services.ReportMultiThreadedBeforeOrAfterMethod.reportMultiThreadedBeforeOrAfterMethod(ReportMultiThreadedBeforeOrAfterMethod.java:74)
2021-09-01T06:02:41.6291907Z 	at io.trino.testng.services.ReportMultiThreadedBeforeOrAfterMethod.onBeforeClass(ReportMultiThreadedBeforeOrAfterMethod.java:42)
2021-09-01T06:02:41.6294776Z 	at org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:167)
2021-09-01T06:02:41.6298729Z 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
2021-09-01T06:02:41.6300081Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2021-09-01T06:02:41.6301473Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2021-09-01T06:02:41.6302407Z 	at java.base/java.lang.Thread.run(Thread.java:829)
```

instead of making the test single-threaded, you can move the mutable state into the (only) test method.